extmod/modure.c

```diff
@@ -174,10 +249,184 @@ STATIC mp_obj_t re_split(size_t n_args, const mp_obj_t *args) {
+                                is_group_number = 1;
+                                end_group = value + 1;
+                            }
+                        }
+                    }
+                } else if (group != 0 && unichar_isdigit(*group)) {
+                    // search group with syntax "\number"
+                    const char *value = group;
+                    int value_l = str_to_int(value, &match_no);
+                    if (match_no > -1 && match_no < match->num_matches) {
+                        is_group_number = 0;
+                        end_group = value += value_l;
+                    }
+                }
+
+                if (match_no > -1 && is_group_number > -1 && start_group != NULL && end_group != NULL) {
+                    const char *start_match = match->caps[match_no * 2];
+                    if (start_match == NULL) {
+                        // no match for this group
+                        return where;
+                    }
+
+                    size_t mg_l = (match->caps[match_no * 2 + 1] - start_match);
+                    size_t gv_l = (end_group - start_group);
+
+                    if (gv_l < mg_l) {
+                        vstr_add_len(vstr_repl, (mg_l - gv_l));
+                        repl_p += gv_l;
+                    } else if (gv_l > mg_l) {
+                        vstr_cut_out_bytes(vstr_repl, (start_group - replace_str), (gv_l - mg_l));
+                        repl_p -= mg_l;
+                    }
+
+                    // replace the substring matched by group
+                    memmove((char *)(start_group + mg_l), (start_group + gv_l), strlen(start_group));
+                    memcpy((char *)(start_group), start_match, mg_l);
+                }
+            }
+        } while (*(++repl_p) != 0);
+
+        // add replace
+        vstr_add_str(vstr_return, vstr_str(vstr_repl));
+
+        // post-match string
+        post_m_l = (match->caps[1] - where_str);
+
+        vstr_free(vstr_repl);
+
+        num_sub++;
+
+        subj.begin = match->caps[1];
+        subj.end = subj.begin + strlen(match->caps[1]);
+
+        if (count > 0 && --count == 0) {
+            break;
+        }
+
+        m_del_var(mp_obj_match_t, char*, match->num_matches, match);
+    }
+
+    // add post-match string
+    if (post_m_l != -1) {
+        vstr_add_str(vstr_return, &where_str[post_m_l]);
+    }
+
+    const mp_obj_type_t *str_type = mp_obj_get_type(where);
+    return (!num_sub ? where : mp_obj_new_str_from_vstr(str_type, vstr_return));
+}
+
+STATIC mp_obj_t re_sub_helper(mp_obj_t self, size_t n_args, const mp_obj_t *args) {
+    mp_obj_t replace = args[1];
+    mp_obj_t where = args[2];
+    mp_int_t count = 0;
+    mp_int_t flags = 0;
+    if (n_args > 3) {
+        count = mp_obj_get_int(args[3]);
+        if (n_args > 4) {
+            flags = mp_obj_get_int(args[4]);
```

Comment:
@dmazzella FYI I removed the keyword argument handling from this function and just made count and flags optionally positional args.

Reply:
ok